src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.cs

```diff
@@ -205,6 +234,22 @@ public static Vector4 HardLight(Vector4 backdrop, Vector4 source)
     private static float OverlayValueFunction(float backdrop, float source)
         => backdrop <= 0.5f ? (2 * backdrop * source) : 1 - (2 * (1 - source) * (1 - backdrop));
 
+    /// <summary>
+    /// Helper function for Overlay and HardLight modes
+    /// </summary>
+    /// <param name="backdrop">Backdrop color element</param>
+    /// <param name="source">Source color element</param>
+    /// <returns>Overlay value</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector256<float> OverlayValueFunction(Vector256<float> backdrop, Vector256<float> source)
+    {
+        Vector256<float> left = Avx.Multiply(Avx.Multiply(Vector256Two, backdrop), source);
```

Comment:
`Multiply(Two, backdrop)` is better as `Add(backdrop, backdrop)`. 

This is something I'm working on recognizing implicitly in the JIT as part of the constant folding support being added in .NET 8

Reply:
You could then consider a helper method here:
```csharp
public static Vector256<float> MultiplyAddEstimate(Vector256<float> left, Vector256<float> right, Vector256<float> addend)
{
    if (Fma.IsSupported)
    {
        return Fma.MultiplyAdd(left, right, addend);
    }
    else
    {
        return Avx.Add(Avx.Multiply(left, right), addend);
    }
}
```

The reason that it's an "estimate" is that `FMA` does a more precise computation and gives a more accurate result. This can also subtly change the answer in some edge cases.

The most notably is that for `double`, `(1e308 * 2) - 1e308` returns `infinity`, but `FusedMultiplyAdd(1e308, 2, -1e308)` returns `1e300` (there is a similar scenario for `float`, but I don't remember the exact inputs off the top of my head).

Most other inputs are less drastic in their differences. They'll either produce the same result (with `Fma` typically being faster) or be off by 1 ULP (if you looked at the raw bits, the least significant bit would typically differ by 1).

Reply:
We actually have a helper that does exactly that in `SimdUtils.HwIntrinsics.MultiplyAdd`

Reply:
https://github.com/dotnet/runtime/issues/82366